### PR TITLE
build(cmake/linux): Use CMake for building on Linux in CI

### PR DESCRIPTION
**Pull request by quyykk**
## Feature Details

This replaces scons with CMake when building on Linux in the CI environment.

**Note:** This deliberately doesn't touch the Steam workflow 

## Testing Done

CI will test everything.

### CODE REVIEW

**Comment by quyykk** [comment]
Release is here: https://github.com/quyykk/endless-sky/releases/tag/v16linux2

I tested it and it works!

File: .github/workflows/cd.yaml [hunk 1]
```diff
@@ -15,30 +15,37 @@ concurrency:
...
     steps:
       - uses: actions/checkout@v3
       - name: Install dependencies
         run: |
           sudo rm /etc/apt/sources.list.d/* && sudo dpkg --clear-avail # Speed up installation and get rid of unwanted lists
-          sudo apt-get update
-          sudo apt-get install -y --no-install-recommends libsdl2-dev libpng-dev libjpeg-turbo8-dev libopenal-dev libmad0-dev libglew-dev libgl1-mesa-dev libegl1-mesa-dev libgles2-mesa-dev uuid-dev
-          python3 -m pip install --user scons
+          sudo apt update
+          sudo apt install -y --no-install-recommends libxmu-dev libxi-dev libgl-dev libglu1-mesa-dev libgles2-mesa-dev libwayland-dev libxkbcommon-dev libegl1-mesa-dev
```
Comment (MCOfficer) [inline .github/workflows/cd.yaml:30]:
```suggestion
          sudo apt-get update
          sudo apt-get install -y --no-install-recommends libxmu-dev libxi-dev libgl-dev libglu1-mesa-dev libgles2-mesa-dev libwayland-dev libxkbcommon-dev libegl1-mesa-dev
```
Should still be `apt-get` for automated invocations

File: .github/workflows/cd_release.yaml [hunk 2]
```diff
@@ -6,36 +6,36 @@ on:
...
           sudo rm /etc/apt/sources.list.d/* && sudo dpkg --clear-avail # Speed up installation and get rid of unwanted lists
-          sudo apt-get update
-          sudo apt-get install -y --no-install-recommends libsdl2-dev libpng-dev libjpeg-turbo8-dev libopenal-dev libmad0-dev libglew-dev libgl1-mesa-dev libegl1-mesa-dev libgles2-mesa-dev uuid-dev
-          python3 -m pip install --user scons
-      - name: Build AppImage
-        run: ./utils/build_appimage.sh
-      - name: Upload artifact
-        uses: actions/upload-artifact@v3
+          sudo apt update
+          sudo apt install -y --no-install-recommends libxmu-dev libxi-dev libgl-dev libglu1-mesa-dev libgles2-mesa-dev libwayland-dev libxkbcommon-dev libegl1-mesa-dev
```
Comment (MCOfficer) [inline .github/workflows/cd_release.yaml:22]:
```suggestion
          sudo apt-get update
          sudo apt-get install -y --no-install-recommends libxmu-dev libxi-dev libgl-dev libglu1-mesa-dev libgles2-mesa-dev libwayland-dev libxkbcommon-dev libegl1-mesa-dev
```
dito

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -57,90 +57,39 @@ jobs:
...
-    - name: Print toolchain versions
-      if: steps.cache-artifact.outputs.cache-hit != 'true'
-      run: |
-        g++ -v
-        gcc-ar --version
-        ld -v
-        scons --version
-        python --version
+        sudo apt update
+        sudo apt install -y --no-install-recommends libxmu-dev libxi-dev libgl-dev libglu1-mesa-dev libgles2-mesa-dev libwayland-dev libxkbcommon-dev libegl1-mesa-dev libosmesa6 mesa-utils libglvnd-dev
```
Comment (MCOfficer) [inline .github/workflows/ci.yml:74]:
```suggestion
        sudo apt-get update
        sudo apt-get install -y --no-install-recommends libxmu-dev libxi-dev libgl-dev libglu1-mesa-dev libgles2-mesa-dev libwayland-dev libxkbcommon-dev libegl1-mesa-dev libosmesa6 mesa-utils libglvnd-dev
```

**Review by MCOfficer — CHANGES_REQUESTED** [review]
What's the reason for yeeting entire test jobs?

**Comment by quyykk** [comment]
> What's the reason for yeeting entire test jobs?

Oh yeah I forgot to add a comment. The only one I removed is the memory leak test, because it's not really useful, it only checks memory errors and leaks during parsing. I'd rather add sanitizers to the integration tests than run valgrind during parsing.

**Comment by petervdmeer** [comment]
When I run the unit-tests through scons, then I get a nice overview of the amount of assertions and test-cases:
```
$ scons test
scons: Reading SConscript files ...
scons: done reading SConscript files.
scons: Building targets ...
Running tests...
===============================================================================
All tests passed (75921 assertions in 60 test cases)

scons: done building targets.
```



While [running the unit-tests through CMake](https://github.com/quyykk/endless-sky/blob/use-cmake-on-linux/readme-cmake.md) does not seem to tell me this:
```
$ ctest --preset linux-test
Test project /home/peter/[...]/endless-sky/build/linux
1/1 Test #71: unit
100% tests passed, 0 tests failed out of 1

Label Time Summary:
unit    =   0.07 sec*proc (1 test)

Total Test time (real) =   0.30 sec
```

Is it possible to also list the amount of assertions and test-cases as is done with Scons?

**Comment by petervdmeer** [comment]
And can we extend the instructions at https://github.com/quyykk/endless-sky/blob/use-cmake-on-linux/readme-cmake.md with a method to run the integration tests on linux without xvfb? (Just directly on the running X-server?)

**Comment by petervdmeer** [comment]
> And can we extend the instructions at https://github.com/quyykk/endless-sky/blob/use-cmake-on-linux/readme-cmake.md with a method to run the integration tests on linux without xvfb? (Just directly on the running X-server?)

Probably best that we describe that in https://github.com/quyykk/endless-sky/blob/use-cmake-on-linux/tests/integration/README.md, and just link to the higher level test-directory from the CMake list of tests. (For more tests, see ../tests/README.md .)

**Comment by quyykk** [comment]
> Is it possible to also list the amount of assertions and test-cases as is done with Scons?

Amount of test cases yes, it would look like this:

```
$ ctest --preset linux-test
Test project /home/user/projects/endless-sky/build/linux
58/58 Test #58: Truncate enum
100% tests passed, 0 tests failed out of 58

Label Time Summary:
unit    =   0.53 sec*proc (58 tests)

Total Test time (real) =   0.57 sec
```

(58 test cases instead of 60 because this is from an older build). But this can only be done once we drop SCons .Alternatively, you can pass `--verbose`. I could enable that by default, but it produces a lot of unnecessary output:

```
$ ctest --preset linux-test --verbose
UpdateCTestConfiguration  from :/home/user/projects/endless-sky/DartConfiguration.tcl
UpdateCTestConfiguration  from :/home/user/projects/endless-sky/build/linux/DartConfiguration.tcl
Parse Config file:/home/user/projects/endless-sky/build/linux/DartConfiguration.tcl
Test project /home/user/projects/endless-sky/build/linux
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 71
    Start 71: unit

71: Test command: /home/user/projects/endless-sky/build/linux/tests/Debug/endless-sky-tests
71: Working Directory: /home/user/projects/endless-sky/tests
71: Test timeout computed to be: 1500
71: ===============================================================================
71: All tests passed (75924 assertions in 60 test cases)
71: 
1/1 Test #71: unit .............................   Passed    0.05 sec

100% tests passed, 0 tests failed out of 1

Label Time Summary:
unit    =   0.05 sec*proc (1 test)

Total Test time (real) =   0.14 sec
``` 

If you run the benchmarks, you'll notice that it uses verbose as well. Unfortunately it's the only way to see the output :frowning: 

> And can we extend the instructions at [quyykk/endless-sky@use-cmake-on-linux/readme-cmake.md](https://github.com/quyykk/endless-sky/blob/use-cmake-on-linux/readme-cmake.md?rgh-link-date=2023-01-21T13%3A32%3A44Z) with a method to run the integration tests on linux without xvfb? (Just directly on the running X-server?)

That's what the debug version of the integration tests is for. I don't think it's useful to see the non-debug integration tests running because it is way too fast.

**Review by petervdmeer — APPROVED** [review]

**Comment by petervdmeer** [comment]
> (58 test cases instead of 60 because this is from an older build). But this can only be done once we drop SCons

I would like to see this done in a follow-up.

**Review by MCOfficer — APPROVED** [review]

**Pull request merged**